src/client/caching/structures/guilds/Guild.ts

```diff
@@ -402,25 +523,219 @@ export class Guild extends Structure {
+ */
+export interface GuildPruneDryOptions {
+	/**
+	 * The number of days to count prune for (1 or more).
+	 * @since 0.0.1
+	 * @default 7
+	 */
 	days: number;
+
+	/**
+	 * The {@link Role role} IDs to include.
+	 * @since 0.0.1
+	 * @default []
+	 */
+	include_roles: string[];
+}
+
+/**
+ * The options for a non-dry {@link Guild#prune}.
+ * @since 0.0.1
+ * @see https://discord.com/developers/docs/resources/guild#begin-guild-prune-query-string-params
+ */
+export interface GuildPruneOptions extends GuildPruneDryOptions {
+	/**
+	 * Whether 'pruned' is returned, discouraged for large guilds.
+	 * @since 0.0.1
+	 * @default !this.large
+	 */
 	compute_prune_count: boolean;
 }
+
+/**
+ * The vanity URL retrieved from {@link Guild#fetchVanityURL}.
+ * @since 0.0.1
+ * @see https://discord.com/developers/docs/resources/guild#get-guild-vanity-url-example-partial-invite-object
+ */
+export interface GuildVanityURL {
+	/**
+	 * The code of this invite.
+	 * @since 0.0.1
+	 * @example "discord"
+	 */
+	code: string;
+
+	/**
+	 * The amount of uses this invite has.
+	 * @since 0.0.1
+	 * @example 42
+	 */
+	uses: number;
+}
+
+/**
+ * The options for the widget image.
+ * @since 0.0.1
+ * @see https://discord.com/developers/docs/resources/guild#get-guild-widget-image-query-string-params
+ */
+export interface WidgetImageOptions {
+	/**
+	 * Style of the widget image returned.
+	 * @since 0.0.1
+	 * @default WidgetStyle.Shield
+	 */
+	style?: WidgetStyle;
+}
+
+/**
+ * The widget style options.
+ * @since 0.0.1
+ * @see https://discord.com/developers/docs/resources/guild#get-guild-widget-image-widget-style-options
+ */
+export enum WidgetStyle {
+	/**
+	 * shield style widget with Discord icon and guild members online count
+	 * @since 0.0.1
+	 * @see https://discord.com/api/guilds/81384788765712384/widget.png?style=shield
+	 */
+	Shield = 'shield',
+
+	/**
+	 * large image with guild icon, name and online count. "POWERED BY DISCORD" as the footer of the widget
+	 * @since 0.0.1
+	 * @see https://discord.com/api/guilds/81384788765712384/widget.png?style=banner1
+	 */
+	Banner1 = 'banner1',
+
+	/**
+	 * smaller widget style with guild icon, name and online count. Split on the right with Discord logo
+	 * @since 0.0.1
+	 * @see https://discord.com/api/guilds/81384788765712384/widget.png?style=banner2
+	 */
+	Banner2 = 'banner2',
+
+	/**
+	 * large image with guild icon, name and online count. In the footer, Discord logo on the left and "Chat Now" on the right
+	 * @since 0.0.1
+	 * @see https://discord.com/api/guilds/81384788765712384/widget.png?style=banner3
+	 */
+	Banner3 = 'banner3',
+
+	/**
+	 * large Discord logo at the top of the widget. Guild icon, name and online count in the middle portion of the widget and a "JOIN MY SERVER" button at the bottom
+	 * @since 0.0.1
+	 * @see https://discord.com/api/guilds/81384788765712384/widget.png?style=banner4
+	 */
+	Banner4 = 'banner4'
+}
```

Comment:
This should be moved to @klasa/rest

Reply:
Feel free to port it, just please don't forget the docs so when we have the website, people have everything documented ❤️